catalog/core/catalog-core-eventcommands/src/main/java/org/codice/ddf/catalog/pubsub/command/ListCommand.java

```diff
@@ -73,46 +80,95 @@
+    columnWidth[1] = entries.stream().mapToInt(s -> s.callbackUrl.length()).max().orElse(-1);
+    columnWidth[2] = entries.stream().mapToInt(s -> s.isEnterprise.length()).max().orElse(-1);
+    columnWidth[3] = entries.stream().mapToInt(s -> s.sourceIds.length()).max().orElse(-1);
+
+    columnWidth[0] = Math.max(columnWidth[0], SUBSCRIPTION_ID_COLUMN_HEADER.length());
+    columnWidth[1] = Math.max(columnWidth[1], CALLBACK_COLUMN_HEADER.length());
+    columnWidth[2] = Math.max(columnWidth[2], ENTERPRISE_COLUMN_HEADER.length());
+    columnWidth[3] = Math.max(columnWidth[3], SOURCE_ID_COLUMN_HEADER.length());
 
-    console.println(
+    String rowFormatString =
         String.format(
-            "%s%s\t %s%s",
-            rowColor,
-            entry.getKey(),
-            entry.getValue().getProperty("event-endpoint"),
-            DEFAULT_CONSOLE_COLOR));
+            "%%-%ds | %%-%ds | %%-%ds | %%-%ds",
+            columnWidth[0], columnWidth[1], columnWidth[2], columnWidth[3]);
+
+    console.println();
+    console.print("Total subscriptions found: " + entries.size());
+    console.println();
+    console.println();
+    console.print(CYAN_CONSOLE_COLOR);
+    console.printf(
+        rowFormatString,
+        SUBSCRIPTION_ID_COLUMN_HEADER,
+        CALLBACK_COLUMN_HEADER,
+        ENTERPRISE_COLUMN_HEADER,
+        SOURCE_ID_COLUMN_HEADER);
+    console.println(DEFAULT_CONSOLE_COLOR);
+
+    for (SubscriptionListEntry entry : entries) {
+      String rowColor = (entry.ping()) ? DEFAULT_CONSOLE_COLOR : RED_CONSOLE_COLOR;
+      console.print(rowColor);
+      console.printf(
+          rowFormatString, entry.id, entry.callbackUrl, entry.isEnterprise, entry.sourceIds);
+      console.println(DEFAULT_CONSOLE_COLOR);
+    }
+  }
+
+  private class SubscriptionListEntry {
+    private String id;
+    private String callbackUrl;
+    private String isEnterprise;
+    private String sourceIds;
+    private Subscription subscription;
+
+    public SubscriptionListEntry(Map.Entry<String, ServiceReference<Subscription>> entry) {
+      Object eventEndpoint = entry.getValue().getProperty("event-endpoint");
+      this.id = entry.getKey();
+      this.callbackUrl = (eventEndpoint == null) ? "" : (String) eventEndpoint;
+      this.subscription = bundleContext.getService(entry.getValue());
+
+      if (subscription == null) {
+        this.isEnterprise = "";
+        this.sourceIds = "";
+      } else {
+        this.isEnterprise = Boolean.toString(subscription.isEnterprise());
+        this.sourceIds =
+            (subscription.getSourceIds() == null)
+                ? "[]"
+                : Arrays.toString(subscription.getSourceIds().toArray());
+      }
+    }
+
+    public boolean ping() {
+      return (subscription != null)
+          && (subscription.getDeliveryMethod() instanceof Pingable)
+          && ((Pingable) subscription.getDeliveryMethod()).ping();
```

Comment:
❓ Technically this condition is different than what it was before. For instance, this will return false (and color the row red) if the subscription isn't an instance of `Pingable` whereas before it would only color the row red if the entry **_was_** an instance of `Pingable` **and** the ping failed. Was this change intentional?

Reply:
That was not intentional. I'll change it to be consistent with the existing behavior